pkg/cloudprovider/providers/azure/azure_blobDiskController.go

```diff
@@ -62,18 +62,16 @@ var (
 	accountsLock = &sync.Mutex{}
 )
 
-func newBlobDiskController(common *controllerCommon) (*BlobDiskController, error) {
-	c := BlobDiskController{common: common}
-
-	// get accounts
-	accounts, err := c.getAllStorageAccounts()
-	if err != nil {
-		klog.Errorf("azureDisk - getAllStorageAccounts error: %v", err)
-		c.accounts = make(map[string]*storageAccountState)
-		return &c, nil
+func (c *BlobDiskController) initStorageAccounts() {
+	if c.accounts == nil {
+		// get accounts
+		accounts, err := c.getAllStorageAccounts()
```

Comment:
I do not see the logic where the timeout was reduced to 1min, do we still want to keep the 10mins timeout (knowing that the initial issue was solved differently) I will leave that decision to you two :)

Reply:
let's keep the original timeout, this issue is already resolved by moving `getAllStorageAccounts` to only invoked by `CreateBlobDisk`.